tests/functional/aws-node-sdk/test/bucket/getCorsResponse.js

```diff
@@ -0,0 +1,1083 @@
+    it('should respond no such bucket if bucket does not exist', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'GET', headers, code: 'NoSuchBucket',
+        headersResponse: null }, done);
+    });
+    it('should return BadRequest for OPTIONS request without origin', done => {
+        const headers = {};
+        methodRequest({ method: 'OPTIONS', headers, code: 'BadRequest',
+        headersResponse: null }, done);
+    });
+    it('should return BadRequest for OPTIONS request without ' +
+    'Access-Control-Request-Method', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'OPTIONS', headers, code: 'BadRequest',
+        headersResponse: null }, done);
+    });
+});
+
+describe('OPTIONS requests with existing bucket', () => {
+    beforeEach(done => {
+        s3.createBucket({ Bucket: bucket, ACL: 'public-read' }, err => {
+            _waitForAWS(done, err);
+        });
+    });
+    afterEach(done => {
+        s3.deleteBucket({ Bucket: bucket }, err => {
+            _waitForAWS(done, err);
+        });
+    });
+
+    it('should allow GET without CORS', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'GET', headers, code: 200,
+        headersResponse: null }, done);
+    });
+    it('should allow HEAD without CORS', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'HEAD', headers, code: 200,
+        headersResponse: null }, done);
+    });
+    it('should return AccessForbidden for OPTIONS request on bucket without ' +
+    'CORSConfiguration', done => {
+        const headers = {
+            'Origin': 'http://www.allowedwebsite.com',
+            'Access-Control-Request-Method': 'GET',
+        };
+        methodRequest({ method: 'OPTIONS', headers, code: 'AccessForbidden',
+        headersResponse: null }, done);
+    });
+
+    describe('allow PUT, POST, DELETE, GET methods and allow only ' +
+    'one origin', () => {
+        const corsParams = {
+            Bucket: bucket,
+            CORSConfiguration: {
+                CORSRules: [
+                    {
+                        AllowedMethods: [
+                            'PUT', 'POST', 'DELETE', 'GET',
+                        ],
+                        AllowedOrigins: [
+                            'http://www.allowedwebsite.com',
+                        ],
+                    },
+                ],
+            },
+        };
+        beforeEach(done => {
+            s3.putBucketCors(corsParams, done);
+        });
+
+        afterEach(done => {
+            s3.deleteBucketCors({ Bucket: bucket }, done);
+        });
+
+        it('send OPTIONS request with allowed origin', done => {
```

Comment:
should respond to OPTIONS request from allowed origin with access control headers